Source/Plugins/CoherenceViewer/CoherenceNode.cpp

```diff
@@ -215,11 +237,20 @@ void CoherenceNode::updateSettings()
     // Seg/win/step/interp - move to params eventually
     interpRatio = 2;
 
+    if (nGroup1Chans > 0)
+    {
+        float newFs = getDataChannel(group1Channels[0])->getSampleRate();
+        if (newFs != Fs)
+        {
+            Fs = newFs;
+            updateDataBufferSize(segLen * Fs);
+        }
+    }
+
```

Comment:
This is the place where `Fs` might change - `updateSettings` is called for all processors whenever any sample rates change.